Building quality apps/Pull requests.md

```diff
@@ -1,19 +1,72 @@
+
+Not sure what to do with the change you're looking at right now? Skip it and come back to it later.
+
+Finished with all the changes in a file? Collapse the file to get it out of your way and make it easier to navigate and see what's left.
+
+After you're done, give it a last couple of re-reads and re-iterations.
+
+## Finally
+
+After reading your description, the reviewer should be able to have a single pass through the 'Files changed' section, without going back and forth hopelessly trying to make sense of the changes and hopelessly trying to build a mental model.
+
+The title should follow the same format as a commit message heading. That is, use the imperative mood, as if giving a command, and keep it short. You can leave the branch name, if you choose to, but it takes 10 seconds to think of something better. The title is far less important than the description, though.
+
+How much time should you invest in writing a PR description? Use your own reasoning on a case-by-case basis, as with everything. For very small PRs, it will take you a couple of minutes. For a small PR, 15 minutes should be sufficient. For a moderate PR, it's perfectly fine to take 30 minutes. For larger PRs, you may need an hour, or more. The return on investment is huge in either case. The more you practise, the faster and better you'll be, just like with any other skill.
+
+I believe that some examples can be useful. I'll use my own PRs, simply because I was trying to hold myself up to a standard. They are by no means perfect. Form your own opinion about them. Pick up what you like, and change what you don't. Try to review them, as a matter of fact. (Please don't leave actual comments.) See how it feels going through the files changed, or going through the commits. See what made it easier for you, and try to figure out what's still missing from my descriptions. Perhaps try to go through the changes without reading the description, as well.
+
+**ConnectionBuddy**:
+
+- [https://github.com/zplesac/android_connectionbuddy/pull/91](https://github.com/zplesac/android_connectionbuddy/pull/91)
+
+**HAK**:
+
+- [https://github.com/infinum/Android-Hak-Ispitivaci/pull/453](https://github.com/infinum/Android-Hak-Ispitivaci/pull/453)
+- [https://github.com/infinum/Android-Hak-Ispitivaci/pull/454](https://github.com/infinum/Android-Hak-Ispitivaci/pull/454)
+- [https://github.com/infinum/Android-Hak-Ispitivaci/pull/481](https://github.com/infinum/Android-Hak-Ispitivaci/pull/481)
+
+**GO**:
+
+- [https://github.com/infinum/Android-MyGO/pull/517](https://github.com/infinum/Android-MyGO/pull/517)
+- [https://github.com/infinum/Android-MyGO/pull/525](https://github.com/infinum/Android-MyGO/pull/525)
+- [https://github.com/infinum/Android-MyGO/pull/526](https://github.com/infinum/Android-MyGO/pull/526)
+- [https://github.com/infinum/Android-MyGO/pull/521](https://github.com/infinum/Android-MyGO/pull/521)
```

Comment:
I like the idea of having this chapter in the public handbook, but if we're going public we should remove links to our private repos 🙂

Reply:
Oh. I totally forgot that this was a public handbook.